apps/resume-web/src/routes/pdf/index.tsx

```diff
@@ -1,19 +1,51 @@
 import { use, useEffect, useRef, useState } from 'react'
 import { Button } from '@resume/ui'
 import { createFileRoute } from '@tanstack/react-router'
-import { CacheContext } from '@providers/globalCache'
+import { CacheContext, CacheKeys } from '@providers/globalCache'
+import { DownloadIcon } from '@resume/icons'
+import { Cache } from '@services/Cache.types'
 
 const worker = new Worker(new URL('./-src/worker.ts', import.meta.url), { type: 'module' })
 
 export const Route = createFileRoute('/pdf/')({
   component: RouteComponent,
 })
 
+// TODO: Rethink, each section have to manage its validations
+function isCacheComplete(cache: Cache<CacheKeys>): boolean {
+  const cacheData = cache.dehydrate();
+  const personalInfo = cacheData[CacheKeys.PersonalInfo];
+  const education = cacheData[CacheKeys.Education];
+  const experience = cacheData[CacheKeys.Experience];
+  const technicalSkills = cacheData[CacheKeys.TechnicalSkills];
+  const contactInfo = cacheData[CacheKeys.ContactInfo];
+
+  return Boolean(
+    personalInfo?.name &&
+    personalInfo?.address &&
+    personalInfo?.city &&
+    personalInfo?.email &&
+    personalInfo?.profession &&
+    personalInfo?.state &&
+    education?.length &&
+    experience?.length &&
+    technicalSkills?.skillList?.length &&
+    contactInfo?.phoneNumber &&
+    contactInfo?.linkedInProfile &&
+    contactInfo?.twitterProfile &&
+    contactInfo?.githubProfile &&
+    contactInfo?.portfolioLink
+  );
+}
```

Comment:
array of values some with every with callback checks if boolean and if array has length

Reply:
solved